### ISSUE TITLE
focal-loss = 87.3365

### ISSUE THREAD

**Issue by liangzimei** [zimenglan-sysu-512/Focal-Loss#1]
when i change softmaxwithloss to focal-loss, in the beginning of training, the loss becomes 87.3365. 
When using softmaxwithloss, there is no such a  problem.
Is the code is still under the testing phase?

**Comment by zimenglan-sysu-512** [comment]
hi @liangzimei,

sorry about that. i just finish the code. i will test it in the next few days.

**Comment by zimenglan-sysu-512** [comment]
hi @liangzimei,

i have fixed the bugs: the loss and the gradient. now it can work. please have a try.

**Comment by zimenglan-sysu-512** [comment]
#2

**Comment by liangzimei** [comment]
@zimenglan-sysu-512 ,  thanks for your reminder.
my train.prototxt is 
layer {
   name "fc3_new_1"
   type: InnerProduct
   ......
}
layer {
   name: "loss_focal"
   type: "FocalLoss"
   bottom: "fc3_new_1"
   bottom: "label"
   top: "loss_focal"
   loss_param{
     normalize: true
     normalization: FULL
    } 
} 
when i update the cpp and cu just now, the loss become nan in the training process.

**Comment by zimenglan-sysu-512** [comment]
hi @liangzimei,

If i try small dataset using pvanet, it's ok, but it encounters the "NAN", when i use larger dataset. I still try to fix it.

**Comment by zimenglan-sysu-512** [comment]
hi @liangzimei,

i fix the NaN problem. i guess that, the op (1 - p_t)^gamma / (1 - p_t) causes the problem, when 1 - p_t is so small.

now you can have a try.
